src/solver/kthvalue/forward_kthvalue.cpp

```diff
@@ -0,0 +1,143 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/kthvalue/problem_description.hpp>
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_view_utils.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/kthvalue/invoke_params.hpp>
+#include <miopen/kthvalue/solvers.hpp>
+#include <miopen/kthvalue.hpp>
+
+namespace miopen {
+
+namespace solver {
+
+namespace kthvalue {
+
+bool IsImprovementOverROCm(const miopen::kthvalue::FwdProblemDescription& problem)
+{
+    TensorDescriptor inputDesc = problem.GetInputDesc();
+    size_t dimSize             = inputDesc.GetLengths()[problem.GetDim()];
+    size_t dimStride           = inputDesc.GetStrides()[problem.GetDim()];
+    size_t dimNum              = inputDesc.GetLengths().size();
+
+    return dimNum >= 2 && dimStride == 1 && dimSize >= 300;
+}
+
+bool KthvalueFwd::IsApplicable(const ExecutionContext& /*context*/,
+                               const miopen::kthvalue::FwdProblemDescription& problem) const
+{
+    if(!IsImprovementOverROCm(problem))
+        return false;
+    if(problem.GetInputDesc().GetSize() > 5)
+        return false;
+    return true;
+}
+
+ConvSolution KthvalueFwd::GetSolution(const ExecutionContext& context,
+                                      const miopen::kthvalue::FwdProblemDescription& problem) const
+{
+    std::ignore = context;
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto input_desc    = problem.GetInputDesc();
+    auto in_dtype      = miopen::GetDataType(input_desc.GetType());
+    auto dtype         = problem.GetOutputDesc().GetType();
+    auto size          = input_desc.GetElementSize();
+    auto dim_size      = input_desc.GetLengths()[problem.GetDim()];
+    size_t output_size = size / dim_size;
+
+    size_t xlocalsize = 256;
+    if(dim_size >= 8192)
+    {
+        xlocalsize = 512;
+    }
```

Comment:
You did this because when dim_size >= 8192, LOCAL_SIZE 512 produced better performance or for other reasons? How much did it improve compared to LOCAL_SIZE 256 in this case?

Reply:
The performance with LOCAL_SIZE 512 is improved. With LOCAL_SIZE 256 the perf with input_size >= 8192 was about 0.8 -> 1.1 compared to ROCm.